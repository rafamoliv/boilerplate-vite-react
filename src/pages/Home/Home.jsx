import React from 'react'
import { useState } from 'react'
import { useDispatch, useSelector } from 'react-redux'

import { Button } from '@/components'
import { setSignout } from '@/store/Auth'
import { setInitialSetting } from '@/store/InitialSettings'

import { Card, Container, Docs, Logo } from './Home.styles'

import { reactLogo } from '@/assets/icons'
import { useFetchReposQuery } from '@/services/api'

const Home = () => {
  const dispatch = useDispatch()
  const { isInitialized, changeResult } = useSelector(
    (state) => state.initialSetting
  )
  const { data = [] } = useFetchReposQuery()
  const [count, setCount] = useState(0)
  console.log('changeResult', changeResult)
  return (
    <Container>
      <div>
        <a href="https://vitejs.dev" rel="noreferrer" target="_blank">
          <Logo alt="Vite logo" src="/vite.svg" />
        </a>
        <a href="https://reactjs.org" rel="noreferrer" target="_blank">
          <Logo alt="React logo" src={reactLogo} />
        </a>
      </div>

      <h1>Boilerplate Vite + React</h1>

      <Docs>Click on the Vite and React logos to learn more</Docs>

      <Card>
        <Button onClick={() => setCount((count) => count + 1)} primary>
          Hook count is {count}
        </Button>
      </Card>

      <div>
        <Button
          onClick={() => {
            alert(`Clicked! Change isInitialized to: ${!isInitialized}`)
            dispatch(setInitialSetting(!isInitialized))
          }}
          primary
        >
          Launch redux
        </Button>
        <Button onClick={() => dispatch(setSignout())}>SIGN OUT</Button>
      </div>

      <div>
        <p>
          <strong>Call redux: </strong>
        </p>
        {data?.map((x, i) => {
          return <p key={i}>{x.name}</p>
        })}
      </div>
    </Container>
  )
}
export default Home
