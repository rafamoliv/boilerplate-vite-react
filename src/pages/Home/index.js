export { default } from './Home'
export { homeEn, homePtBr } from './locales'
