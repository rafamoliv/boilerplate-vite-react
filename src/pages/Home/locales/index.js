export { default as homePtBr } from './ptBr'
export { default as homeEn } from './en'
