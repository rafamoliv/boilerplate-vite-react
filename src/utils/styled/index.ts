import { css } from 'styled-components'

export const getColors = (prop1, prop2) => {
  return ({ theme }) => css`
    ${theme.colors[prop1][prop2]};
  `
}
