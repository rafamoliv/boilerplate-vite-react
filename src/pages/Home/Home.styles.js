import styled from 'styled-components'

export const Container = styled.div`
  max-width: 80rem;
  margin: ${({ theme }) => theme.spacing[0]} auto;
  padding: ${({ theme }) => theme.spacing[2]};
  text-align: center;
`

export const Logo = styled.img`
  height: ${({ theme }) => theme.spacing[14]};
  margin: ${({ theme }) => theme.spacing[3]};
  display: none;

  &:hover {
    filter: drop-shadow(0 0 2em #646cffaa);
  }

  @media (min-width: ${({ theme }) => theme.breakpoints.sm}) {
    display: inline-block;
  }
`

export const Card = styled.div`
  padding: ${({ theme }) => theme.spacing[4]};
`

export const Docs = styled.p`
  color: #888;
  font-size: ${({ theme }) => theme.font.sizes['xl']};
`
