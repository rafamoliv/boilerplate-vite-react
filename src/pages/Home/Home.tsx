import { faArrowLeftLong } from '@fortawesome/pro-thin-svg-icons'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import * as React from 'react'
import { useTranslation } from 'react-i18next'
import { useDispatch } from 'react-redux'

import { Button } from '@/components'
import { setSignout } from '@/store/Auth'
import { setInitialSetting } from '@/store/InitialSettings'

import { Card, Container, Docs, Logo } from './Home.styles'

import { reactLogo } from '@/assets/icons'
import { useFetchReposQuery } from '@/services/api'
import { useAppSelector } from '@/utils/hooks'

const Home = () => {
  const dispatch = useDispatch()
  const { t } = useTranslation('pgHome')
  const { isInitialized, changeResult } = useAppSelector(
    (state) => state.initialSetting
  )
  const { data = [] } = useFetchReposQuery()
  const [count, setCount] = React.useState(0)

  return (
    <Container>
      <div>
        <a href="https://vitejs.dev" rel="noreferrer" target="_blank">
          <Logo alt="Vite logo" src="/vite.svg" />
        </a>
        <a href="https://reactjs.org" rel="noreferrer" target="_blank">
          <Logo alt="React logo" src={reactLogo} />
        </a>
      </div>

      <h1>{t('title')}</h1>
      <FontAwesomeIcon icon={faArrowLeftLong} />
      <Docs>{t('summary')}</Docs>
      <p>{changeResult}</p>

      <Card>
        <Button onClick={() => setCount((count) => count + 1)} primary>
          {t('count', { value: count })}
        </Button>
      </Card>

      <div>
        <Button
          onClick={() => {
            alert(`Clicked! Change isInitialized to: ${!isInitialized}`)
            dispatch(setInitialSetting(!isInitialized))
          }}
          primary
        >
          {t('btn', { context: 'redux' })}
        </Button>
        <Button onClick={() => dispatch(setSignout())}>
          {t('btn', { context: 'signout' })}
        </Button>
      </div>

      <div>
        <p>
          <strong>{t('redux')}</strong>
        </p>
        {data?.map((x, i) => {
          return <p key={i}>{x.name}</p>
        })}
      </div>
    </Container>
  )
}
export default Home
