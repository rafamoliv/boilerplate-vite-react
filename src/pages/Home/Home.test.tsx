import { screen } from '@testing-library/react'

import { render } from '@/utils/tests'

import Home from './Home'

it('should test', () => {
  render(<Home />)
  expect(
    screen.getByRole('heading', { name: 'Boilerplate Vite + React' })
  ).toMatchSnapshot()
})
