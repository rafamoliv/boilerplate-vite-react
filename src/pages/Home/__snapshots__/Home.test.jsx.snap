// Vitest Snapshot v1

exports[`should test 1`] = `
@media (min-width:48rem) {

}

<h1>
  Boilerplate Vite + React
</h1>
`;
